app/src/main/java/com/example/beatbox/BeatBox.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.beatbox
+
+import android.content.res.AssetManager
+import android.util.Log
+import java.io.IOException
+
+class BeatBox(private val assets: AssetManager) {
+    val sounds: List<Sound>
+
+    init {
+        sounds = loadSounds()
+    }
+
+    private fun loadSounds(): List<Sound> {
+        val soundNames: Array<String>?
+        return try {
+            soundNames = assets.list(SOUNDS_FOLDER)
```

Comment:
This would be much nicer with functional programming.
If you want to use it:
```           
 assets.list(SOUNDS_FOLDER)
                ?.map { fileName ->
                    "$SOUNDS_FOLDER/$fileName"
                }?.map { assetPath ->
                    Sound(assetPath)
                } ?: emptyList()```